Proofs/SHA512/SHA512_block_armv8_rules.lean

```diff
@@ -134,12 +134,16 @@ theorem concat_of_rsh_is_msb_128 (x y : BitVec 64) :
   (x ++ y) >>> 64 = BitVec.zeroExtend 128 x := by
   bv_check "lrat_files/Sha512_block_armv8_rules.lean-concat_of_rsh_is_msb_128-101-2.lrat"
 
--- (FIXME) Generalize to arbitrary-length bitvecs.
 theorem truncate_of_concat_is_lsb_64 (x y : BitVec 64) :
   BitVec.zeroExtend 64 (x ++ y) = y := by
   bv_check "lrat_files/Sha512_block_armv8_rules.lean-truncate_of_concat_is_lsb_64-106-2.lrat"
 
--- (FIXME) Generalize to arbitrary-length bitvecs.
+theorem zeroExtend_append_eq (w : Nat) (x y : BitVec w) :
+  BitVec.zeroExtend w (x ++ y) = y := by
+  ext
+  simp only [truncate_append, Nat.le_refl, ↓reduceDIte, zeroExtend_eq]
```

Comment:
The style in LNSym is currently a bit inconsistent, but standard Lean/Mathlib style is to indent the type/statement of a theorem with 4 spaces (vs 2 spaces for the proof itself).

Then, we can generalize the theorem further: append is a heterogenous operation and can take two bitvectors of different length! Since the value `w` and `v` is determined by the other arguments `x` and `y`, we can make the former implicit arguments.

Finally, you could consider naming this `zeroExtend_append_eq_right`, to draw extra attention to the fact that the value we're "zero-extending" to is exactly the width of the right operand, but it's also perfectly valid to argue that's the logical statement for a theorem about zeroExtend and append, and keep preferring the shorter name, I'll leave that choice up to you.

```suggestion
/-- truncating the result of appending two bitvectors to the width of the right operand gives
exactly the right operand -/
theorem zeroExtend_append_eq {w v : Nat} (x : BitVec w) (y : BitVec v) :
    BitVec.zeroExtend v (x ++ y) = y := by
  ext
  simp only [truncate_append, Nat.le_refl, ↓reduceDIte, zeroExtend_eq]
```